File: week2/w2d1/binary_search.py
xs = [1,2,3,4,5]
def search(xs, number):
    minimum = 0
    maximum = len(xs) - 1  
    while (minimum <= maximum):
        average = int((minimum+maximum) /2)
        if (xs[average] == number):
            return average
        elif (xs[average] < number):
            minimum = average + 1
        else:
            maximum = average - 1
            
    return -1
print(search(xs, 5))

File: java_script/3_py_to_js/lab1/binary_search.js
let xs = [1, 3, 4, 6, 7, 9];
alert(xs)
let x = prompt("What number do you want to find? ")

function binary_search(xs, x) {
	let min = 0;
	let max = xs.length-1;
	while (min <= max){
        const average = Math.floor((min+max)/2);
        if (xs[average] == x) {
            return average;
        }
        
        else if (xs[average] < x) {
            min = average + 1;
        
        }

        else {
            max = average -1;
        }
   
    
    }
    return 'not existed';
}
alert(binary_search(xs, x))